File: app/src/main/java/com/github/pockethub/android/ui/item/dialog/LabelDialogItem.java
package com.github.pockethub.android.ui.item.dialog;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.CheckBox;
import android.widget.TextView;

import com.github.pockethub.android.R;
import com.github.pockethub.android.ui.issue.LabelDrawableSpan;
import com.github.pockethub.android.ui.item.BaseDataItem;
import com.github.pockethub.android.ui.item.BaseViewHolder;
import com.meisolsson.githubsdk.model.Label;

import butterknife.BindView;

public class LabelDialogItem extends BaseDataItem<Label, LabelDialogItem.ViewHolder> {

    private boolean selected;

    public LabelDialogItem(Label dataItem, boolean selected) {
        super(null, dataItem, dataItem.name().hashCode());
        this.selected = selected;
    }

    @Override
    public void bind(@NonNull ViewHolder viewHolder, int position) {
        LabelDrawableSpan.setText(viewHolder.name, getData());
        viewHolder.selected.setChecked(selected);
    }

    @Override
    public int getLayout() {
        return R.layout.label_item;
    }

    public void toggleSelected() {
        selected = !selected;
    }

    public boolean isSelected() {
        return selected;
    }

    @NonNull
    @Override
    public ViewHolder createViewHolder(@NonNull View itemView) {
        return new ViewHolder(itemView);
    }


    class ViewHolder extends BaseViewHolder {
        @BindView(R.id.tv_label_name) TextView name;
        @BindView(R.id.cb_selected) CheckBox selected;

        public ViewHolder(@NonNull View rootView) {
            super(rootView);
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/dialog/LabelDialogItem.kt
package com.github.pockethub.android.ui.item.dialog

import com.github.pockethub.android.R
import com.github.pockethub.android.ui.issue.LabelDrawableSpan
import com.meisolsson.githubsdk.model.Label
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.label_item.*

class LabelDialogItem(
        val label: Label,
        selected: Boolean
) : Item(label.name()!!.hashCode().toLong()) {

    var isSelected: Boolean = selected
        private set

    override fun getLayout() = R.layout.label_item

    override fun bind(holder: ViewHolder, position: Int) {
        LabelDrawableSpan.setText(holder.tv_label_name, label)
        holder.cb_selected.isChecked = isSelected
    }

    fun toggleSelected() {
        isSelected = !isSelected
    }
}
